### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by StratusFearMe21**
I reduced the number of allocations made by this program overall, and I updated the versions of all of the libraries used by this program. I also switched from `termion` to `crossterm` to add partial Windows compatibility

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +0,0 @@
-[submodule "aur/2048-rs"]
```
Comment (pierrechevalier83) [inline .gitmodules]: Was this removal intentional? If so, what's the motivation for it?

**Review by pierrechevalier83 — CHANGES_REQUESTED** [review]
Thanks for the PR, it's a nice refactor, and I like the idea to use `crossterm` to enable cross-platform functionality. I'm just wondering whether these 2 commits from @jackpot51 from 2017 were committed intentionally as part of this PR. If not, could you rebase your branch to exclude them? I'm OK with updating the cargo.lock, but I'd rather not remove the aur submodule without a reason for it.

File: .gitmodules (hunk 1 above)
Comment (StratusFearMe21) [reply .gitmodules]: Whoops, that removal was not intentional

**Comment by StratusFearMe21** [comment]
Sorry about that, I started this PR from the repo on https://gitlab.redox-os.org/redox-os/2048-rs

**Review by pierrechevalier83 — APPROVED** [review]
Thanks so much for the refactoring :smile: LGTM

**Pull request merged**